### PR TITLE
[DT-161][risk=no] removing basics, overall health and lifestyle data

### PR DESCRIPTION
**Pull request by freemabd**
Removing basics, overall health and lifestyle data when all participant questions have a null questionnaire response id

### CODE REVIEW

File: api/db-cdr/generate-cdr/build-search-all-events.sh [hunk 1]
```diff
@@ -388,7 +388,9 @@ WHERE o.observation_concept_id is not null
               where domain_id = 'Observation'
               and vocabulary_id in ('CPT4')
               and standard_concept = 'S'
-            )"
+            )
+    and c.vocabulary_id != 'PPI'
+    and c.concept_class_id != 'Survey'"
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-search-all-events.sh:393]: Found that we were inserting standard rows for surveys. This is not needed, so added check to stop this.

File: api/db-cdr/generate-cdr/build-search-all-events.sh [hunk 2]
```diff
@@ -457,4 +459,199 @@ FROM \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` v
...
+"DELETE
+ FROM \`$BQ_PROJECT.$BQ_DATASET.cb_search_all_events\` se
+ WHERE EXISTS (
+   WITH lifestyle_survey_remove_participants AS (
+     SELECT person_id, observation_date, observation_source_concept_id, questionnaire_response_id
+     FROM \`$BQ_PROJECT.$BQ_DATASET.R2022Q4R3.observation\`
+     WHERE person_id IN (
+         SELECT DISTINCT person_id
+         FROM \`$BQ_PROJECT.$BQ_DATASET.observation\`
+         WHERE (observation_source_concept_id IN (903581, 1384450)
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-search-all-events.sh:606]: these concept ids represent some deprecated questions on the basics survey

File: api/db-cdr/generate-cdr/build-search-all-events.sh [hunk 3]
```diff
@@ -457,4 +459,199 @@ FROM \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` v
...
+             AND type = 'PPI'
+             AND subtype = 'QUESTION'
+         ))
+         AND questionnaire_response_id IS NULL
+
+           EXCEPT DISTINCT
+
+         SELECT DISTINCT person_id
+         FROM \`$BQ_PROJECT.$BQ_DATASET.observation\`
+         WHERE (observation_source_concept_id IN (903581, 1384450)
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-search-all-events.sh:625]: these concept ids represent some deprecated questions on the basics survey

File: api/db-cdr/generate-cdr/build-search-all-events.sh [hunk 4]
```diff
@@ -457,4 +459,199 @@ FROM \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` v
...
+             FROM \`$BQ_PROJECT.$BQ_DATASET.cb_criteria\`
+             WHERE concept_id IN (1586134)
+               AND domain_id = 'SURVEY' ) a ON (c.path LIKE CONCAT('%', a.id, '.%'))
+           WHERE domain_id = 'SURVEY'
+             AND type = 'PPI'
+             AND subtype = 'QUESTION'
+         ))
+         AND questionnaire_response_id IS NOT NULL
+     )
+     AND (observation_source_concept_id IN (903581, 1384450)
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-search-all-events.sh:640]: these concept ids represent some deprecated questions on the basics survey

Comment (chenchals) [reply api/db-cdr/generate-cdr/build-search-all-events.sh:640]: LGTM

**Review by chenchals — APPROVED** [review]
LGTM. Hopefully this wont be a recurring theme...

**Pull request merged**